src/configs/FirebaseConfig.js

```diff
@@ -77,6 +78,25 @@ const FirebaseConfig = (props) => {
     }
   };
 
+  const getUserProfileV3 = async (url) => {
+    if (url && typeof url === 'string') {
+      const userId = await getUserId();
+      const parts = url.split('?username=');
+      const username = parts?.[1];
+
+      try {
+        const replacedUsername = username?.replace('+', '');
+        const data = {
+          username: replacedUsername,
+          user_id: userId
+        };
+        handleMovePage(USER, data);
+      } catch (e) {
+        console.error(e);
+      }
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `getUserProfileV3` function, there is no error handling for the case when the username is not found in the URL. This could lead to unexpected behavior if the URL does not contain a `username` parameter after `?username=`. Also, the replacement of '+' with '' may not be necessary or correct depending on the expected format of the username. If usernames can include '+', this replacement would alter the username incorrectly. Consider adding error handling for missing username and re-evaluating the need for replacing '+'.
```diff
     const getUserProfileV3 = async (url) => {
       if (url && typeof url === 'string') {
         const userId = await getUserId();
         const parts = url.split('?username=');
         const username = parts?.[1];
 
+        if (!username) {
+          console.error('Username not found in URL');
+          return;
+        }
+
         try {
-          const replacedUsername = username?.replace('+', '');
+          const replacedUsername = username;
           const data = {
             username: replacedUsername,
             user_id: userId
           };
           handleMovePage(USER, data);
         } catch (e) {
           console.error(e);
         }
       }
     };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->